plugins/nextcloud/NextcloudAddressBook.php

```diff
@@ -1,90 +1,129 @@
 <?php
+
 use RainLoop\Providers\AddressBook\Classes\Contact;
 
 class NextcloudAddressBook implements \RainLoop\Providers\AddressBook\AddressBookInterface
 {
 	use RainLoop\Providers\AddressBook\CardDAV;
-	private const URI = 'webmail';
 	private $contactsManager;
-	private $key;
 
-	function __construct()
+	private const SETTINGS_KEY = 'nextCloudAddressBookUri';
+
+	private string $defaultUri = 'webmail';
+	private string $defaultName = 'WebMail';
+	private string $defaultDescription = 'Recipients from snappymail';
+	private bool $ignoreSystemAddressBook = true;
+
```

Comment:
contactsManager is not declated here

Reply:
it is at line 8... refactoring it so it can easily visible.

Reply:
@akhil1508 please check again